src/utils/middleware/rateLimiter.ts

```diff
@@ -0,0 +1,29 @@
+import { NextFunction, Request, Response } from "express";
+import {
+  RateLimiterRes,
+  RateLimiterStoreAbstract
+} from "rate-limiter-flexible";
+
+import { log } from "../../utils/logger";
+
+export const makeRateLimiterMiddleware = (
+  rateLimiter: RateLimiterStoreAbstract
+) => (req: Request, res: Response, next: NextFunction) => {
+  const rateLimiterKey = req.ip;
+  return rateLimiter
+    .consume(rateLimiterKey)
+    .then(_ => next())
```

Comment:
Could be useful setting RateLimit headers even on authorized API calls.

Reply:
what do you mean exactly ?

Reply:
```suggestion
    .then(_ => {
      res
        .set("X-RateLimit-Remaining", _.remainingPoints.toString())
        .set(
          "X-RateLimit-Reset",
          new Date(Date.now() + Number(_.msBeforeNext)).toString()
        )
        .set(
          "X-RateLimit-Limit",
          (_.remainingPoints + _.consumedPoints).toString()
        );
      next();
    })
```
So any API call response has the RateLimit infos that could eventually used by the client.
[Here](https://opensource.zalando.com/restful-api-guidelines/#153) some reference on API guidelines

Reply:
Thank you now I get it ! Good catch !